### PR TITLE
Cicd

### PR DESCRIPTION
**Pull request by pholica**
Add:
 * Pull request check running testsuite
 * On-merge deploy action if deploy credentials are set

### CODE REVIEW

**Comment by pholica** [comment]
I believe this could be it. The lftp can work with URLs in form of ftp://username:password@host/path/to/dest

**Comment by pholica** [comment]
Ok, it seems there are some things that need to be fixed for python3. Also, I'm not sure if and how well the `push` github action works.

**Comment by pholica** [comment]
The on push github workflow works in my fork, I just don't have access to the runner so the actual push code is not yet tested.

**Review by jkonecny12 — COMMENTED** [review]
Just a few notes.

File: .github/workflows/push_deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        id: push_ftp_check
+        run: |
+            echo "::set-output name=push_ftp_url_available::${{ secrets.PUSH_FTP_URL != '' }}"
+
+  deploy-image:
+    needs: [secrets-check]
+    runs-on: [self-hosted, kstest]
+    container:
+      image: quay.io/centos/centos:stream9
+    if: needs.secrets-check.outputs.push_ftp_url_available == 'true'
```
[the remark this answers is not in the document]
Comment (pholica) [reply .github/workflows/push_deploy.yaml:25]: If I'm not mistaken the if statement desn't have access to secrets as described here: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-secrets

I believe that if the documented approach was used, the env would not be set if the runner was not available => the if statement wouldn't be evaluated but I'm just guessing here based on @velezd findings.

[the remark this answers is not in the document]
Comment (jkonecny12) [reply .github/workflows/push_deploy.yaml:23]: I was thinking that this could be the reason. Unfortunately, these issues sometimes happen with GH actions.

**Comment by pholica** [comment]
Ok, everything works as expected on my end now. Let's continue with the review :)

**Review by velezd — APPROVED** [review]
Looks good, thanks

**Pull request merged**